uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/models/absence-type.model.js

```diff
@@ -61,6 +63,24 @@ define([
           .then(function (results) {
             return results.length > 0;
           });
+        },
+        /**
+         * Retrieves calculation units
+         * and sets units symbols to provided absence types accordingly
+         *
+         * Example: if a unit name is "hours", then the symbol will be "h"
+         *
+         * @param   {Array} absenceTypes array of absence types or their instances
+         * @return  {Promise} resolves with the input populated with calculation unit symbols
+         */
+        loadCalculationUnits: function (absenceTypes) {
+          return OptionGroup.valuesOf('hrleaveandabsences_absence_type_calculation_unit')
+            .then(function (calculationUnits) {
+              return _.map(absenceTypes, function (absenceType) {
+                return _.assign(absenceType, { calculation_unit_symbol:
+                  _.indexBy(calculationUnits, 'value')[absenceType.calculation_unit].name[0]});
```

Comment:
I'd suggest moving the `_.indexBy` outside of `_.map`, because you are creating an index for each absence type, which is not optimal, there should only be one index.